source/core_json.c

```diff
@@ -1248,7 +1248,7 @@ static JSONStatus_t search( char * buf,
                             size_t * outValueLength )
 {
     JSONStatus_t ret = JSONPartial;
-    size_t i = 0, key, keyLength, value, valueLength;
+    size_t i = 0, key, keyLength, value = 0, valueLength = 0;
```

Comment:
what abt ```key``` and ```keyLength```?

Reply:
There are no warnings for key and keyLength.  I think it is better to enlist the compiler to tell me when I use an uninitialized variable, than to initialize everything and rob myself of that check.  The only problem is when the compiler gets it wrong, as the Microsoft compiler does here.

Reply:
Do you think that it's good practice to just initialize everything anyways as part of our style?

Reply:
No, not everything.  I think the practice can do more harm than good in some cases.